File: app/src/main/java/com/qltech/bws/FaqModule/Models/FaqListModel.java
package com.qltech.bws.FaqModule.Models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class FaqListModel implements Parcelable {
    @SerializedName("ResponseData")
    @Expose
    private ArrayList<ResponseData> responseData = null;
    @SerializedName("ResponseCode")
    @Expose
    private String responseCode;
    @SerializedName("ResponseMessage")
    @Expose
    private String responseMessage;
    @SerializedName("ResponseStatus")
    @Expose
    private String responseStatus;

    protected FaqListModel(Parcel in) {
        responseCode = in.readString();
        responseMessage = in.readString();
        responseStatus = in.readString();
    }

    public static final Creator<FaqListModel> CREATOR = new Creator<FaqListModel>() {
        @Override
        public FaqListModel createFromParcel(Parcel in) {
            return new FaqListModel(in);
        }

        @Override
        public FaqListModel[] newArray(int size) {
            return new FaqListModel[size];
        }
    };

    public ArrayList<ResponseData> getResponseData() {
        return responseData;
    }

    public void setResponseData(ArrayList<ResponseData> responseData) {
        this.responseData = responseData;
    }

    public String getResponseCode() {
        return responseCode;
    }

    public void setResponseCode(String responseCode) {
        this.responseCode = responseCode;
    }

    public String getResponseMessage() {
        return responseMessage;
    }

    public void setResponseMessage(String responseMessage) {
        this.responseMessage = responseMessage;
    }

    public String getResponseStatus() {
        return responseStatus;
    }

    public void setResponseStatus(String responseStatus) {
        this.responseStatus = responseStatus;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(responseCode);
        parcel.writeString(responseMessage);
        parcel.writeString(responseStatus);
    }

    public static class ResponseData implements Parcelable{
        @SerializedName("ID")
        @Expose
        private String iD;
        @SerializedName("Title")
        @Expose
        private String title;
        @SerializedName("Desc")
        @Expose
        private String desc;
        @SerializedName("VideoURL")
        @Expose
        private String videoURL;
        @SerializedName("Category")
        @Expose
        private String Category;

        protected ResponseData(Parcel in) {
            iD = in.readString();
            title = in.readString();
            desc = in.readString();
            videoURL = in.readString();
            Category = in.readString();
        }

        public static final Creator<ResponseData> CREATOR = new Creator<ResponseData>() {
            @Override
            public ResponseData createFromParcel(Parcel in) {
                return new ResponseData(in);
            }

            @Override
            public ResponseData[] newArray(int size) {
                return new ResponseData[size];
            }
        };

        public String getID() {
            return iD;
        }

        public void setID(String iD) {
            this.iD = iD;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }

        public String getDesc() {
            return desc;
        }

        public void setDesc(String desc) {
            this.desc = desc;
        }

        public String getVideoURL() {
            return videoURL;
        }

        public void setVideoURL(String videoURL) {
            this.videoURL = videoURL;
        }

        public String getCategory() {
            return Category;
        }

        public void setCategory(String planFlag) {
            Category = planFlag;
        }

        @Override
        public int describeContents() {
            return 0;
        }

        @Override
        public void writeToParcel(Parcel parcel, int i) {
            parcel.writeString(iD);
            parcel.writeString(title);
            parcel.writeString(desc);
            parcel.writeString(videoURL);
            parcel.writeString(Category);
        }
    }
}

File: app/src/main/java/com/brainwellnessspa/FaqModule/Models/FaqListModel.kt
package com.brainwellnessspa.FaqModule.Models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class FaqListModel protected constructor(`in`: Parcel) : Parcelable {
    @SerializedName("ResponseData")
    @Expose
    var responseData: ArrayList<ResponseData>? = null

    @SerializedName("ResponseCode")
    @Expose
    var responseCode: String?

    @SerializedName("ResponseMessage")
    @Expose
    var responseMessage: String?

    @SerializedName("ResponseStatus")
    @Expose
    var responseStatus: String?
    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(responseCode)
        parcel.writeString(responseMessage)
        parcel.writeString(responseStatus)
    }

    class ResponseData protected constructor(`in`: Parcel) : Parcelable {
        @SerializedName("ID")
        @Expose
        var iD: String?

        @SerializedName("Title")
        @Expose
        var title: String?

        @SerializedName("Desc")
        @Expose
        var desc: String?

        @SerializedName("VideoURL")
        @Expose
        var videoURL: String?

        @SerializedName("Category")
        @Expose
        var category: String?
        override fun describeContents(): Int {
            return 0
        }

        override fun writeToParcel(parcel: Parcel, i: Int) {
            parcel.writeString(iD)
            parcel.writeString(title)
            parcel.writeString(desc)
            parcel.writeString(videoURL)
            parcel.writeString(this.category)
        }

        companion object {
            @JvmField val CREATOR: Parcelable.Creator<ResponseData?> = object : Parcelable.Creator<ResponseData?> {
                override fun createFromParcel(`in`: Parcel): ResponseData? {
                    return ResponseData(`in`)
                }

                override fun newArray(size: Int): Array<ResponseData?> {
                    return arrayOfNulls(size)
                }
            }
        }

        init {
            iD = `in`.readString()
            title = `in`.readString()
            desc = `in`.readString()
            videoURL = `in`.readString()
            this.category = `in`.readString()
        }
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<FaqListModel?> = object : Parcelable.Creator<FaqListModel?> {
            override fun createFromParcel(`in`: Parcel): FaqListModel? {
                return FaqListModel(`in`)
            }

            override fun newArray(size: Int): Array<FaqListModel?> {
                return arrayOfNulls(size)
            }
        }
    }

    init {
        responseCode = `in`.readString()
        responseMessage = `in`.readString()
        responseStatus = `in`.readString()
    }
}